op/model.py

```diff
@@ -237,6 +294,63 @@ def __init__(self, backend):
     def _load(self):
         return self._backend.config_get()
 
+class Status(ABC):
+    """Status values specific to applications and units."""
+
+    _statuses = {}
+
+    def __init__(self, message):
+        self.message = message
+
+    def __new__(cls, *args, **kwargs):
+        if cls is Status:
+            raise TypeError("cannot instantiate a base class")
+
+        return super().__new__(cls)
+
+    @classmethod
+    def _register_status(cls, name, type_):
+        """For use by subclasses only."""
+        cls._statuses[name] = type_
+
+    @classmethod
+    def from_string(cls, name, message):
+        return cls._statuses[name](message)
+
+    def __init_subclass__(cls):
+        super().__init_subclass__()
+        Status._register_status(cls.name, cls)
+
+class ActiveStatus(Status):
+    """The unit believes it is correctly offering all the services it has been asked to offer."""
+    name = 'active'
+
+    def __init__(self, message=''):
```

Comment:
The parameter shouldn't exist, so people get an actual error if they try to setup a message where one makes no sense.

Reply:
Still pending as well.